src/Types.jl

```diff
@@ -88,18 +88,46 @@ Base.:(==)(r1::GitRepo, r2::GitRepo) =
     r1.source == r2.source && r1.rev == r2.rev && r1.subdir == r2.subdir
 
 
-Base.@kwdef mutable struct PackageSpec
-    name::Union{Nothing,String} = nothing
-    uuid::Union{Nothing,UUID} = nothing
-    version::Union{Nothing,VersionTypes,String} = VersionSpec()
-    tree_hash::Union{Nothing,SHA1} = nothing
-    repo::GitRepo = GitRepo()
-    path::Union{Nothing,String} = nothing
-    pinned::Bool = false
+mutable struct PackageSpec
+    name::Union{Nothing,String}
+    uuid::Union{Nothing,UUID}
+    version::Union{Nothing,VersionTypes,String}
+    tree_hash::Union{Nothing,SHA1}
+    repo::GitRepo
+    path::Union{Nothing,String}
+    pinned::Bool
     # used for input only
-    url = nothing
-    rev = nothing
-    subdir = nothing
+    url
+    rev
+    subdir
+    function PackageSpec(; name::Union{Nothing,String} = nothing,
```

Comment:
This will now remove the default constructors so perhaps this should be an outer one?

Reply:
Good call, yeah I've moved it to be an outer constructor now.